app/app.go

```diff
@@ -1548,7 +1548,13 @@ func (d *Deployments) UpdateDeviceDeploymentStatus(ctx context.Context, deployme
 
 	if !ddState.Status.Active() {
 		l := log.FromContext(ctx)
-		if err := d.db.SaveLastDeviceDeploymentStatus(ctx, *dd); err != nil {
+		ldd := model.DeviceDeployment{
+			DeviceId:     dd.DeviceId,
+			DeploymentId: dd.DeploymentId,
+			Id:           dd.Id,
+			Status:       ddState.Status,
+		}
+		if err := d.db.SaveLastDeviceDeploymentStatus(ctx, ldd); err != nil {
```

Comment:
this is already happening in enterprise, smart to put it here up front.

Reply:
previously we were using just *dd here, so we didn't have last status